src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,269 @@
+
+    catch (error) {
+      if (error.statusCode === 404) {
+        // if doesn't exist - create
+        const entityId = await this.createDataFolder(userprofile, token)
+        if (!entityId) {
+          throw ('no folder')
+        }
+        result.dataFolderId = entityId;
+      } else {
+        this.onError(error)
+        return Promise.reject(error);
+      }
+    }
+    finally {
+      // get the file listing
+      result.fileList = await this.getFileListingInFolder(token, result.dataFolderId)
+      this.setState({
+        userprofile,
+        dataFolderId: result.dataFolderId,
+        fileList: result.fileList || [],
+        isLoading: false
+      })
+      return result;
+    }
+  }
+  onError = (args: any) => {
+    console.log(args)
+  }
+
+
+  // creates data folder for user
+  createDataFolder = async (userprofile: UserProfile, token: string): Promise<string | undefined> => {
+
+    const newFolder: Entity = {
+      parentId: this.props.parentContainerId,
+      name: userprofile.ownerId,
+      concreteType: 'org.sagebionetworks.repo.model.Folder',
+    }
+    try {
+      const folder: Entity = await SynapseClient.createEntity(newFolder, token);
+      return folder.id
+    }
+    catch (error) {
+      this.onError(error)
+      throw (error)
+    }
+  }
+
+
+
+  getFileListingInFolder = async (token: string, targetFolderId: string): Promise<Entity[]> => {
+    const request = {
+      includeTypes: ['file'],
+      parentId: targetFolderId,
+      sortBy: 'NAME',
+      sortDirection: 'ASC'
+    }
+
+    try {
+      const data = await SynapseClient.getEntityChildren(request, token);
+      return data.page;
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+  }
+
+  deleteFile = async (token: string, entityId: string): Promise<any> => {
+
+    this.setState({
+      isLoading: true, modalContext: undefined
+    })
+
+    try {
+      await SynapseClient.deleteEntity(token, entityId);
+      const fileList = await this.getFileListingInFolder(token, this.state.dataFolderId!)
+      this.setState({
+        fileList
+      })
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+    finally {
+      this.setState({
+        isLoading: false
+      })
+    }
+  }
+
+  /* ------------------------------------------   rendering fns  ------------------------------------------------*/
+ 
+  renderLoading = (token: string | undefined, isLoading: boolean): JSX.Element => {
+    if (
+      token &&
+      isLoading) {
+      return (<div className='text-center'>
+        <span>Loading&hellip;</span>
+        <span style={{ marginLeft: '2px' }} className={'spinner'} />
+      </div>)
+    } else {
+      return <></>
+    }
+  }
+
+  renderFileTable = (fileList: Entity[], pathpart: string, dataFolderId?: string): JSX.Element => {
+    if (!dataFolderId) {
+      this.onError('Data Folder ID is undefined')
+      return <></>
+    }
+    if (_.isEmpty(fileList)) {
```

Comment:
is there a reason why you can't check fileList.length === 0 for this?

Reply:
fewer keystrokes:) I will change it though